shm/src/Travel.cpp

```diff
@@ -0,0 +1,63 @@
+#include "Travel.hpp"
+
+#include <iostream>
+
+#include "TimeServiceLocator.hpp"
+#include "map.hpp"
+#include "player.hpp"
+
+Travel::Travel(Map* map, Time* time)
+    : map_(map), time_(time), coordinate_x(-1), coordinate_y(-1) {}
+
+void Travel::execute(Player* player) {
+    displayCurrentPosition();
+    displayPossibleOptions();
+    enterCoordinates();
+    while (!isChosenIslandValid()) {
+        std::cout << "Chosen island is not valid!\n";
+        enterCoordinates();
+    }
+    auto daysOfTravel = countDaysOfTravel(player->getSpeed());
+    incrementDays(daysOfTravel);
+    changeCurrentPositionToChosenIsland();
+    std::cout << "Time of travel: " << daysOfTravel << "\n";
+}
+
+void Travel::displayPossibleOptions() {
+    for (auto elem : map_->getAllIslands()) {
+        if (elem.getCoordinates() == map_->getCurrentPosition()->getCoordinates()) {
+            continue;
+        }
+        std::cout << elem << "\n";
+    }
+}
+
+void Travel::enterCoordinates() {
+    std::cout << "Enter the coordinates of the chosen island: ";
+    std::cin >> coordinate_x >> coordinate_y;
+}
+
+void Travel::displayCurrentPosition() {
+    std::cout << "Your current position: " << map_->getCurrentPosition()->getCoordinates() << "\n";
+}
+
+void Travel::incrementDays(size_t count) {
+    for (size_t i = 0; i < count; ++i) {
+        TimeServiceLocator::getTime()->operator++();
```

Comment:
Hmm this is somewhat suspicious - did you ensure, that during travel days are also changed on different objects ( i mean we are calling next day method on other observers) - we want to travel in space, not in time :)

Reply:
I have checked that - its Ok.